src/transformers/models/fuyu/image_processing_fuyu.py

```diff
@@ -1,44 +1,195 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Image processor class for Fuyu."""
+
 import math
-from typing import List, Union, Dict
+from typing import Dict, List, Optional, Union
 
 import numpy as np
 
-from ...image_processing_utils import BaseImageProcessor
+from ...image_processing_utils import BaseImageProcessor, BatchFeature
 from ...image_transforms import (
-    normalize,
     pad,
     resize,
+    to_channel_dimension_format,
 )
 from ...image_utils import (
     ChannelDimension,
+    ImageInput,
+    PILImageResampling,
     get_image_size,
     infer_channel_dimension_format,
     is_scaled_image,
+    is_valid_image,
+    make_list_of_images,
     to_numpy_array,
-    make_list_of_images
 )
-from ...utils import is_torch_available, is_vision_available, logging, requires_backends
+from ...utils import (
+    TensorType,
+    is_torch_available,
+    is_torch_device,
+    is_torch_dtype,
+    is_vision_available,
+    logging,
+    requires_backends,
+)
 
 
 if is_vision_available():
-    import PIL
+    pass
 
 if is_torch_available():
     import torch
 
 logger = logging.get_logger(__name__)
 
 
+def make_list_of_list_of_images(
+    images: Union[List[List[ImageInput]], List[ImageInput], ImageInput]
+) -> List[List[ImageInput]]:
+    if is_valid_image(images):
+        return [[images]]
+
+    if isinstance(images, list):
+        return [make_list_of_images(image) for image in images]
+
+    if isinstance(images, list) and all(isinstance(image, list) for image in images):
+        return images
```

Comment:
```suggestion
    if isinstance(images, list) and all(isinstance(image, list) for image in images):
        return images

    if isinstance(images, list):
        return [make_list_of_images(image) for image in images]
```

shouldn't these conditions be reversed?

Reply:
Yep - sorry 🤦‍♀️